lib/shopify_transporter/exporters/magento/soap.rb

```diff
@@ -74,13 +80,19 @@ def soap_client
         end
 
         def soap_session_id
-          @soap_session_id ||= soap_client.call(
+          return @soap_session_id if @soap_session_id.present?
+
+          result = soap_client.call(
             :login,
             message: {
               username: @username,
               api_key: @api_key,
             }
-          ).body[:login_response][:login_return]
+          ).body
```

Comment:
Given we do the same digging twice, I think this would be preferable:
```ruby
     returned_response = result&.dig(:login_response, :login_return)
     raise FailedLoginError, result.to_s unless returned_response.present?
     @soap_session_id ||= returned_response

Reply:
@JordanVerasamy This is non-blocking but would be great if u could fix it.

Reply:
^^ ignore this, github screwed me over ; _ ;